src/main/java/book/store/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,62 @@
+package book.store.service.impl;
+
+import book.store.dto.cartitem.CartItemRequestDto;
+import book.store.dto.shoppingcart.ShoppingCartDto;
+import book.store.exception.EntityNotFoundException;
+import book.store.mapper.CartItemMapper;
+import book.store.mapper.ShoppingCartMapper;
+import book.store.model.CartItem;
+import book.store.model.ShoppingCart;
+import book.store.repository.cartitem.CartItemRepository;
+import book.store.repository.shoppingcart.ShoppingCartRepository;
+import book.store.service.ShoppingCartService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
```

Comment:
```suggestion
    @Override
    @Transactional
```

Check other places where is neede